src/remapper.rs

```diff
@@ -0,0 +1,648 @@
+/// Module helping in high level copy/move operations across different
```

Comment:
Would you mind adding an example showing what this looks like?

Reply:
I've added two examples, also with a song writing (which is more of an overwriting, as everything is not properly reversed yet).

Reply:
By "everything is not properly reversed yet" do you mean that serialization is not complete? If so, would you mind adding some TODOs to the readme describing what is missing?